cfme/services/catalogs/service_catalogs.py

```diff
@@ -33,47 +39,74 @@
     ])
 
 
-menu.nav.add_branch(
-    'services_catalogs',
-    {
-        'service_catalogs':
-        [
-            lambda _: accordion.click('Service Catalogs'),
-            {
-                'service_catalog':
-                [
-                    lambda ctx: accordion_tree(
-                        'All Services', ctx['catalog'], ctx['catalog_item'].name),
-                    {
-                        'order_service_catalog': lambda _: sel.click(order_button)
-                    }
-                ]
-            }
-        ]
-    }
-)
-
-
-class ServiceCatalogs(Updateable, Pretty):
+match_page = partial(match_location, title='Catalogs', controller='catalog')
+
+
+def nav_to_all():
+    from cfme.web_ui.menu import nav
+    nav._nav_to_fn('Services', 'Catalogs')(None)
+    tree = accordion.tree('Service Catalogs')
+    tree.click_path('All Services')
+
+
+class ServiceCatalogs(Updateable, Pretty, Navigatable):
     pretty_attrs = ['service_name']
 
-    def __init__(self, service_name=None, stack_data=None):
+    def __init__(self, service_name=None, stack_data=None, appliance=None):
         self.service_name = service_name
         self.stack_data = stack_data
+        Navigatable.__init__(self, appliance=appliance)
 
-    def order(self, catalog, catalog_item):
-        sel.force_navigate('order_service_catalog',
-                           context={'catalog': catalog,
-                                    'catalog_item': catalog_item})
+    def order(self):
+        navigate_to(self, 'Order')
         sel.click(form_buttons.submit)
         wait_for(flash.get_messages, num_sec=10, delay=2)
         flash.assert_success_message("Order Request was Submitted")
 
-    def order_stack_item(self, catalog, catalog_item):
-        sel.force_navigate('order_service_catalog',
-                           context={'catalog': catalog,
-                                    'catalog_item': catalog_item})
+    def order_stack_item(self):
+        navigate_to(self, 'Order')
         stack_form.fill(self.stack_data)
```

Comment:
This looks like the only difference between the order and order_stack_item functions at this point, it would be nice to condense this into a single `order` function that takes a boolean `fill` that triggers calling `stack_form.fill()`. I don't think its a change that needs to be made in this PR, but otherwise the code is duplicated in each function.